src/intrinsics.cpp

```diff
@@ -305,9 +305,13 @@ static Value *emit_unbox(Type *to, const jl_cgval_t &x, jl_value_t *jt)
     if (jt == (jl_value_t*)jl_bool_type)
         return builder.CreateZExt(builder.CreateTrunc(builder.CreateLoad(p), T_int1), to);
 
-    if (!x.isboxed) // stack has default alignment
+    if (x.isboxed)
+        return builder.CreateAlignedLoad(p, 16); // julia's gc gives 16-byte aligned addresses
```

Comment:
Yes, the alignment is too large if GC is not guaranteeing 16-byte alignment for small objects.  For that matter, providing an alignment bigger than the object seems unlikely to provide any benefit, except for  a few highly contrived cases.  (Exercise left to the student :-).  Can someone provide a concise summary of the GC alignment guarantees?  E.g., a function that maps "object size" x "object LLVM alignment" -> "min guaranteed alignment in Julia heap"?